pkg/client/client_test.go

```diff
@@ -1910,7 +1946,303 @@ func TestClose(t *testing.T) {
 
 }
 
+type ClientTestSuiteTrackEvent struct {
+	suite.Suite
+	mockProcessor       *MockProcessor
+	mockDecisionService *MockDecisionService
+	client              OptimizelyClient
+}
+
+func (s *ClientTestSuiteTrackEvent) SetupTest() {
+	mockProcessor := new(MockProcessor)
+	s.mockProcessor = mockProcessor
+	s.mockDecisionService = new(MockDecisionService)
+
+	s.client = OptimizelyClient{
+		ConfigManager:      ValidProjectConfigManager(),
+		DecisionService:    s.mockDecisionService,
+		EventProcessor:     s.mockProcessor,
+		NotificationCenter: notification.NewNotificationCenter(),
+	}
+}
+
+func (s *ClientTestSuiteTrackEvent) TestTrackWithNotification() {
+
+	s.mockProcessor.On("ProcessEvent", mock.Anything).Return(true)
```

Comment:
`mock.Anything` is very general. We are using ConversionEvent, so please use that type.